### DISCUSSION TITLE
Replace OOB Sharepoint top menu with Top Application Customizer Extension

### DISCUSSION THREAD

**Discussion by asadshaikhcodes** [SharePoint/sp-dev-docs#7962] [Q&A] [answered]
I have built a custom Sharepoint Top Application Customizer Extension for a branding project, does the Top Application Customizer always appears on the top of the page? as per our requirement we are manipulating the css on the page and hiding the OOB menu, and replacing it with our custom menu extension which I feel a workaround or hack. Is there a way to disable the OOB menu programmatically? also how the Top Application Customizer works with the Hub site? please help me on the better way of implementation of the Top Application Customizer.

**Comment by AJIXuMuK** [comment, marked as answer]
@asadshaikhcodes - SPFx Application Customizer always appears on the top of the page, there is no way to configure its location.

> as per our requirement we are manipulating the css on the page and hiding the OOB menu, and replacing it with our custom menu extension which I feel a workaround or hack

That is definitely a workaround that may break in future. Any DOM manipulations, including CSS, are not supported. We can change the markup of the page without a notice.

I would suggest removing navigation using settings that you can find under Change the Look -> Navigation menu option:
![image](https://user-images.githubusercontent.com/17036219/169082982-08de3636-5398-4cf0-8a95-0ee8c711f441.png)

>  also how the Top Application Customizer works with the Hub site?

The customizer is displayed on top of hub navigation.
